zbench.zig

```diff
@@ -84,7 +93,37 @@ pub const Benchmark = struct {
     ) !void {
         try self.benchmarks.append(self.allocator, Definition{
             .name = name,
-            .func = func,
+            .defn = .{ .simple = func },
+            .config = optional(Config, config, self.common_config),
+        });
+    }
+
+    /// Add a benchmark function to be timed with `run()`.
+    pub fn addP(
```

Comment:
The `addP` name is maybe too terse, just I couldn't think of anything not unpleasantly long.

Reply:
You mean something like this would be too long? `addParametricBenchmark` or `addParamBenchmark`. I have no strong opinion here, from my side we can leave it as it is.

Reply:
addParamBenchmark strikes a nice balance IMHO, and is much more descriptive than addP. 

 addParameterized or addParametric or addParam are some more options

Reply:
Maybe `addParam`, there's no need to mention "benchmark" in the `add` method so no need here I think.